test/e2e/upgrades/upgrade.go

```diff
@@ -41,6 +41,9 @@ const (
 
 // Test is an interface for upgrade tests.
 type Test interface {
+	// Name should return a test name sans spaces.
+	Name() string
```

Comment:
What about using reflection to get the name instead? Or just key the framework array off the test poniters?